Tests/NextcloudIntegrationTests/FilesIntegrationTests.swift

```diff
@@ -0,0 +1,88 @@
+            XCTAssertEqual(NKError.success.errorCode, error.errorCode)
+            XCTAssertEqual(NKError.success.errorDescription, error.errorDescription)
+
+            Thread.sleep(forTimeInterval: 0.2)
+
+            // Test reading folder, should exist
+            NCNetworking.shared.readFolder(serverUrl: serverUrlFileName, account: self.user) { account, metadataFolder, metadatas, metadatasUpdate, metadatasLocalUpdate, metadatasDelete, error in
+                XCTAssertEqual(self.account, account)
+                XCTAssertEqual(NKError.success.errorCode, error.errorCode)
+                XCTAssertEqual(NKError.success.errorDescription, error.errorDescription)
+                XCTAssertEqual(metadataFolder?.fileName, folderName)
+
+                // Check Realm directory, should exist
+                let directory = NCManageDatabase.shared.getTableDirectory(predicate: NSPredicate(format: "serverUrl == %@", serverUrlFileName))
+                XCTAssertNotNil(directory)
+
+                Thread.sleep(forTimeInterval: 0.2)
+
+                Task {
+                    // Test deleting folder
+                    await _ = NCNetworking.shared.deleteMetadata(metadataFolder!, onlyLocalCache: false)
+
+                    XCTAssertEqual(NKError.success.errorCode, error.errorCode)
+                    XCTAssertEqual(NKError.success.errorDescription, error.errorDescription)
+
+                    try await Task.sleep(for: .milliseconds(200))
+
+                    // Test reading folder, should NOT exist
+                    NCNetworking.shared.readFolder(serverUrl: serverUrlFileName, account: self.user) { account, metadataFolder, metadatas, metadatasUpdate, metadatasLocalUpdate, metadatasDelete, error in
+                        defer { expectation.fulfill() }
+
+                        XCTAssertEqual(404, error.errorCode)
+                        XCTAssertNil(metadataFolder?.fileName)
+
+                        // Check Realm directory, should NOT exist
+                        let directory = NCManageDatabase.shared.getTableDirectory(predicate: NSPredicate(format: "serverUrl == %@", serverUrlFileName))
+                        XCTAssertNil(directory)
+                    }
+                }
+            }
+        }
+        
+        waitForExpectations(timeout: 100)
+    }
+
+    func test_uploadChunkedFile_withProperParams_shouldUploadChunksAndAssembleFile() {
+//        let metadata = NCManageDatabase.shared.createMetadata(account: account, user: user, userId: userId, fileName: "testFile", fileNameView: fileName, ocId: NSUUID().uuidString, serverUrl: serverUrl, urlBase: uploadAssets.userBaseUrl.urlBase, url: "", contentType: "", isLivePhoto: livePhoto)
+//        NCNetworking.shared.uploadChunkedFile(metadata: tableMetadata.init()) { error in
+//            <#code#>
+//        }
+    }
```

Comment:
I tried to make a test for this, but I can't seem to find a way to upload a file without using UI. Any suggestions here would be welcome @marinofaggiana. Not sure if it's even possible.